plugins/filter/merge.py

```diff
@@ -88,7 +88,18 @@ def merge_with_database(base, database, app_name, customer, environment):
 
     result = deepcopy(base)
 
-    database_name = "_".join([environment[0], customer, app_name])
+    mapping_env = {
+        "ci": "c",
+        "development": "d",
+        "feature": "f",
+        "production": "p",
+        "staging": "s",
+        "preprod": "t",
+    }
+
+    env_type = mapping_env.get(environment) if environment in mapping_env else environment[0]
```

Comment:
You can write it like this
```python
 env_type = mapping_env.get(environment, environment[0])
```

This being said, I think this could create unexpected issues if the environment is not in your mapping but starts with the same letter as one of your environments? 

Should we make it configurable and raise an issue if the environment is not known?

Reply:
Yes this can lead to an environment having the same letter. I don't know how ta make it configurable. If we want to make it configurable we have to inject in the filter I think (I will check the documentation) and I'm not really comfortable with this.

Reply:
or (better):

```
# tasks/create_app_database_vault.yml
databases: "{{ databases | default({}) | merge_with_database(item, app.name, customer, env_types | json_query(name=env_type) }}"
```

The latest parameter will be a `dict` with the environment name and code.

Reply:
:+1: